addon-self-administration-plugin/src/main/java/org/osiam/addons/self_administration/plugin/impl/PluginImpl.java

```diff
@@ -92,37 +92,43 @@ private AccessToken getAccessToken() {
 
     private OsiamConnector getOsiamConnector() {
         if(connector == null){
-            connector = new OsiamConnector.Builder()
-                .withEndpoint(getOsiamEndpoint())
-                .setClientId(getClientId())
-                .setClientSecret(getClientSecret())
-                .build();
+            OsiamConnector.Builder builder = new OsiamConnector.Builder()
+                    .setClientId(getClientId())
+                    .setClientSecret(getClientSecret());
+
+            if (getOsiamVersion().startsWith("2")) {
+                builder.setEndpoint(getOsiamEndpoint());
+            } else {
+                builder.withEndpoint(getOsiamEndpoint());
+            }
+
+            connector = builder.build();
         }
 
         return connector;
     }
 
-    private String getProperty(String key, String defaultValue) {
-        return System.getProperty(key, defaultValue);
-    }
-
     private String getUserName() {
-        return getProperty("osiam.addon-self-administration.plugin.user.name", "admin");
+        return System.getProperty("osiam.addon-self-administration.plugin.user.name", "admin");
     }
 
     private String getUserPassword() {
-        return getProperty("osiam.addon-self-administration.plugin.user.password", "koala");
+        return System.getProperty("osiam.addon-self-administration.plugin.user.password", "koala");
     }
 
     private String getOsiamEndpoint() {
-        return getProperty("osiam.addon-self-administration.plugin.osiam.endpoint", "http://localhost:8080/osiam");
+        return System.getProperty("osiam.addon-self-administration.plugin.osiam.endpoint", "http://localhost:8080/osiam");
+    }
+
+    private String getOsiamVersion() {
+        return System.getProperty("osiam.addon-self-administration.plugin.osiam.version", "3");
```

Comment:
Declaring a version in a code file as string literal is not really considered best practice. But I see that much more indirection (like reading from `pom`) is not suitable here. If you are sure that this is the "correct" way in this case, I'm ok with that.